engine/src/flutter/lib/web_ui/lib/src/engine/web_paragraph/paragraph.dart

```diff
@@ -468,13 +467,17 @@ class WebTextStyle implements ui.TextStyle {
   bool hasElement(StyleElements element) {
     switch (element) {
       case StyleElements.background:
-        return background != null;
+        // Transparent background is equivalent to no background
+        // We do not check for transparency in other paints (like foreground) because
+        // it seems unnatural to have a transparent paint on them
+        return background != null && background!.color != const ui.Color(0x00000000);
```

Comment:
`ui.Color(0x00000000)` is not the only transparent color:

```suggestion
        return background != null && background!.color.a != 0;
```